src/microbe_stage/systems/EngulfingSystem.cs

```diff
@@ -1459,6 +1461,9 @@ private void EjectEverythingFromDeadEngulfer(ref Engulfer engulfer, in Entity en
 
         RemoveEngulfedObject(ref engulfer, engulfableObject, ref engulfable, false);
 
+        ref var damageCooldown = ref entity.Get<DamageCooldown>();
+        damageCooldown.StartCooldown(Constants.PILUS_MAX_COOLDOWN);
```

Comment:
For safety I would add:

```
if (entity.HasComponent<DamageCooldown>())
{
    ...
}
```

here.

Also did you run the threaded run generation script? Having this system now access DamageCooldown may impact the valid execution orders.

Reply:
No, I didn't. Do I need to run this tool every time I've added some metadata to a system?

Reply:
Yes, that's correct because otherwise the old system run order will be used, and it may be incorrect due to the new system run requirements.